plugins/env/app/models/external_environment_variable_group.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+require 'aws-sdk-s3'
+
+class ExternalEnvironmentVariableGroup < ActiveRecord::Base
+  S3_URL_REGEX = /\Ahttps:\/\/([^.]+)\.s3\.amazonaws\.com\/([\w\W]+)\Z/i.freeze
+  S3_URL_FORMAT = "https://[bucket].s3.amazonaws.com/[key]?versionId=[version_id]"
+  HELP_TEXT = ENV.fetch(
+    "EXTERNAL_ENV_GROUP_HELP_TEXT",
+    "Use external service to manage environment variable groups"
+  ).html_safe
+  attr_accessor :key, :bucket, :version_id
+  audited
+  default_scope -> { order(:name) }
+
+  belongs_to :project, inverse_of: :external_environment_variable_groups
+  validates :name, :url, presence: true
+  validates :url, format: {
+    with: S3_URL_REGEX,
+    message: "Invalid format, valid url format is #{S3_URL_FORMAT}"
+  }
+  validate :validate_s3_url
+  after_find :resolve_s3_url
```

Comment:
prefer defining methods for each component and making them call `resolve_s3_url`, always doing certrain work ater finding is usually a bad pattern